static/app/utils/replays/hooks/useDiscoveryQuery.tsx

```diff
@@ -25,14 +24,14 @@ interface Params {
   ignoreCursor?: boolean;
 }
 
-interface State {
-  data: ReplayError[] | undefined;
+interface State<T> {
+  data: T[] | undefined;
   error: Error | undefined;
   isLoading: boolean;
   pageLinks: string | undefined;
 }
 
-const INITIAL_STATE: State = {
+const INITIAL_STATE: State<any> = {
```

Comment:
Remove the `as const` thing just below

It isn't doing anything because we're stomping on the type here. To get the composed effect wrap in Readonly<>:

```suggestion
const INITIAL_STATE: Readonly<State<any>> = {
```


[ts playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAtgTwMLmjAvDA3gCwKYA2BIAXDAEQDuIATgQCbkC+MAhhDKJFANwCwAKEFQEABzwwAylFZQ8AHgAqAPgzZ8RUjEVN+QgVzSJpsvGRNz50GgEswAc1WZchYmSq0GzPYMGHYiABKeKz04AQIFmYwwaHhCPJRVlC2DspO6q5aHnSMur76QA)

<img width="607" alt="Screen Shot 2022-06-15 at 10 43 08 AM" src="https://user-images.githubusercontent.com/187460/173891481-e39de3ce-c4c8-45b7-aecf-9057115ce914.png">